docs/design_doc.md

```diff
@@ -0,0 +1,443 @@
+-   `Tables` CAN NOT be annotated by other `Tables`.
+
+#### Images
+
+Images of a sample. Should conform to the [OME-NGFF concept of an image](https://ngff.openmicroscopy.org/latest/#image-layout).
+
+Images are ndimensional arrays where each element of an array is a pixel of an image. These arrays have labelled dimensions which correspond to:
+
+-   Spatial dimensions (height and width).
+-   Imaging or feature channels.
+-   Z-stacks.
+
+We will support also time-point axes in the future. Furthermore, due to NGFF specs v0.5, such axes will not have name constraints (although they do for first iteration due to NGFF specs v0.4).
+
+The image object itself builds on prior art in image analysis, in particular the [xarray library](https://docs.xarray.dev/en/stable/).
+
+Images have labelled dimensions, and coordinate transformations. These transformations are used to map between pixel space and physical space, and between different physical spaces.
+
+For computational efficiency, images can have a pyramidal or multiscale format. This is implemented as an [xarray datatree](https://github.com/xarray-contrib/datatree).
+
+#### Regions of interest
+
+Regions of interest define distict regions of space that can be used to select and aggregate observations. Regions can correspond to
+
+-   Tissues
+-   Tissue structures
+-   Clinical annotations
+-   Multi-cellular communities
+-   Cells
+-   Subcellular structures
+-   Physical structures from the assay (e.g. visium spots)
+-   Synthetic regions created by analysts (e.g. output of algorithms)
+
+Regions can be used for:
+
+-   subsetting observations (e.g., get all observations in a given region)
+-   aggregating observations (e.g., count all observations in an region)
+
+Regions can be defined in multiple ways.
+
+##### Labels (pixel mask)
+
+Labels are a pixel mask representation of regions. This is an array of integers where each integer value corresponds to a region of space. This is commonly used along side pixel based imaging techniques, where the label array will share dimensionality with the image array. These may also be hierarchichal.
+Should conform to the [OME-NGFF definition](https://ngff.openmicroscopy.org/latest/#image-layout).
+
+The label object itself builds on prior art in image analysis, in particular the [xarray library](https://docs.xarray.dev/en/stable/).
+
+For computational efficiency, labels can have a pyramidal or multiscale format. This is implemented as an [xarray datatree](https://github.com/xarray-contrib/datatree).
+
+##### Polygons
+
+A set of (multi-)polygons associated with a set of observations. Each set of polygons is associated with a coordinate system. Polygons can be used to represent a variety of regions of interests, such as clinical annotations and user-defined regions of interest.
+
+The Polygon object is implemented as a geopandas dataframe with [multi-polygons series](https://geopandas.org/en/stable/docs/user_guide/data_structures.html).
```

Comment:
Furtehr to be added:
`The coordinate system and transforms are stored in ``geopandas.DataFrame.attrs``

Reply:
same as above for labels and images, we store it in xarray.DataArray.attrs

Reply:
Done!